### PR TITLE
[sycl-post-link][NFC] Put passes and their dependencies in namespace llvm

### PR DESCRIPTION
**Pull request by psamolysov-intel**
Currently the SYCLDeviceLibReqMask and SpecConstantsPass classes are declared
inside the global namespace. To make them aligned with all the other declarations
used in the sycl-post-link tool, the passes and their dependencies have been moved
into the llvm namespace.

### CODE REVIEW

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h [hunk 1]
```diff
@@ -33,6 +37,7 @@ enum class DeviceLibExt : std::uint32_t {
 };
 
 using SYCLDeviceLibFuncMap = std::unordered_map<std::string, DeviceLibExt>;
+
 class SYCLDeviceLibReqMaskPass : public ModulePass {
```
Comment (mlychkov) [inline llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:41]: It's not a to-do comment.
@jinge90 @AlexeySachkov I'm just wondering why this scanning was implemented as a pass?
Might it be simpler to define it as a helper function?

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h [hunk 2]
```diff
@@ -33,6 +37,7 @@ enum class DeviceLibExt : std::uint32_t {
 };
 
 using SYCLDeviceLibFuncMap = std::unordered_map<std::string, DeviceLibExt>;
```
Comment (mlychkov) [inline llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:39]: I think it could be moved to cpp file because it doesn't used anywhere else.

Comment (psamolysov-intel) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h]: I've moved the definition to the `.cpp` file and in the anonymous namespace.

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h (hunk 1 above)
Comment (AlexeySachkov) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:39]: > @jinge90 @AlexeySachkov I'm just wondering why this scanning was implemented as a pass?

I'm honestly don't know the history of this pass.

> Might it be simpler to define it as a helper function?

That makes sense to me, we don't really need a pass here, but rather a separate function, which produces a certain result from an input LLVM IR module.

**Review by AlexeySachkov — APPROVED** [review]

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h [hunk 3]
```diff
@@ -17,8 +17,12 @@
 #pragma once
 
 #include "llvm/Pass.h"
+
+#include <cstdint>
+#include <string>
```
Comment (mlychkov) [inline llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:22]: This header may be moved to cpp as well.

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h [hunk 4]
```diff
@@ -17,8 +17,12 @@
...
 #include <unordered_map>
```
Comment (mlychkov) [inline llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:23]: These headers may be moved to cpp as well.

File: llvm/tools/sycl-post-link/SpecConstants.h [hunk 5]
```diff
@@ -46,28 +45,29 @@ struct SpecConstantDescriptor {
...
+
+class SpecConstantsPass : public PassInfoMixin<SpecConstantsPass> {
 public:
   // SetValAtRT parameter controls spec constant lowering mode:
   // - if true, it is lowered to SPIRV intrinsic which retrieves constant value
   // - if false, it is replaced with C++ default (used for AOT compilers)
   SpecConstantsPass(bool SetValAtRT = true) : SetValAtRT(SetValAtRT) {}
-  llvm::PreservedAnalyses run(llvm::Module &M,
-                              llvm::ModuleAnalysisManager &MAM);
+  PreservedAnalyses run(Module &M, llvm::ModuleAnalysisManager &MAM);
```
Comment (mlychkov) [inline llvm/tools/sycl-post-link/SpecConstants.h:57]:
```suggestion
  PreservedAnalyses run(Module &M, ModuleAnalysisManager &MAM);
```

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h (hunk 1 above)
Comment (mlychkov) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h:39]: Submitted an issue https://github.com/intel/llvm/issues/5362.

File: llvm/tools/sycl-post-link/SpecConstants.h (hunk 5 above)
Comment (psamolysov-intel) [reply llvm/tools/sycl-post-link/SpecConstants.h]: @mlychkov Thank you, I missed this.

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h (hunk 4 above)
Comment (psamolysov-intel) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.h]: Thank you, I've moved the headers to .cpp.

**Review by mlychkov — APPROVED** [review]

**Pull request merged**